File: XMLApp/Model/Flight.cs
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;
using XMLApp.Attributes;

namespace XMLApp.Model
{
    [BsonCollection("flights")]
    public class Flight : Document
    {
        private Flight flight;

        public Flight()
        {
        }

        public Flight(Flight flight)
        {
            this.flight = flight;
        }
        public GeoLocation? TakeOffLocation { get; set; }
        public GeoLocation? LandingLocation { get; set; }
        public DateTime TakeOffDate { get; set; }
        public DateTime LandingDate { get; set; }
        public Ticket? Ticket { get; set; }
        public string TicketId { get; set; }
      
    }
}

File: xml-front-app/src/app/model/flight.ts
export class Flight {
    TakeOffLocation: string = ''
    LandingLocation: string = ''
    TakeOffDate: string=''
    LandingDate : string = ''

    public constructor(obj?: Flight) {
        if (obj) {
          this.TakeOffLocation = obj.TakeOffLocation
          this.LandingLocation = obj.LandingLocation
          this.TakeOffDate = obj.TakeOffDate
          this.LandingDate = obj.LandingDate
        }
      }
}